tests/test_inheritance_and_pydantic_generation/test_geting_pydantic_models.py

```diff
@@ -183,7 +184,20 @@ def test_getting_pydantic_model_self_ref():
     assert len(InnerSelf2.__fields__) == 2
     assert set(InnerSelf2.__fields__.keys()) == {"id", "name"}
 
-    assert InnerSelf2 != InnerSelf
+
+def test_getting_pydantic_model_self_ref_exclude():
+    PydanticSelfRef = SelfRef.get_pydantic(exclude={"children": {"name"}})
+    assert len(PydanticSelfRef.__fields__) == 4
+    assert set(PydanticSelfRef.__fields__.keys()) == {
+        "id",
+        "name",
+        "parent",
+        "children",
+    }
+    PydanticSelfRefChildren = PydanticSelfRef.__fields__["children"].type_
+    assert len(PydanticSelfRefChildren.__fields__) == 1
+    assert set(PydanticSelfRefChildren.__fields__.keys()) == {"id"}
+    assert PydanticSelfRef != PydanticSelfRefChildren
```

Comment:
Here you should check both `parent` and `children` and here they should be different

```python
InnerSelf = PydanticSelfRef.__fields__["parent"].type_
assert len(InnerSelf.__fields__) == 2
assert set(InnerSelf.__fields__.keys()) == {"id", "name"}

InnerSelf2 = PydanticSelfRef.__fields__["children"].type_
assert len(InnerSelf2.__fields__) == 1
assert set(InnerSelf2.__fields__.keys()) == {"id"}

assert InnerSelf2 != InnerSelf
```